File: day_trader/sources/strategy/strategies/day_trader/strategy/services/websocket/common/DTO/positions/position_dto.py
from sources.strategy.strategies.day_trader.business_entities.day_trading_position import *
import json

class PositionDTO:
    def __init__(self, dayTradingPos):
        self.Msg="DayTradingPosition"
        self.PosId = dayTradingPos.Id
        self.Symbol = dayTradingPos.Security.Symbol
        self.Active=dayTradingPos.Active
        self.PositionSize = dayTradingPos.SharesQuantity
        self.IsOpen=dayTradingPos.Open()
        self.IsRouting = dayTradingPos.Routing
        self.LongSignal = dayTradingPos.LongSignal
        self.ShortSignal = dayTradingPos.ShortSignal
        self.SignalType = dayTradingPos.SignalType
        self.SignalDesc = dayTradingPos.SignalDesc

        self.RSI=dayTradingPos.MinuteNonSmoothedRSIIndicator.RSI
        self.DailyRSI = dayTradingPos.DailyRSIIndicator.RSI

        self.NetShares = dayTradingPos.GetNetOpenShares()

        if dayTradingPos.GetNetOpenShares()>0:
            self.Direction = "LONG"
        elif dayTradingPos.GetNetOpenShares()<0:
            self.Direction="SHORT"
        else:
            self.Direction="LIQUID"

        self.CurrentMarketPrice = dayTradingPos.MarketData.Trade if dayTradingPos.MarketData is not None else None
        self.CurrentProfit = dayTradingPos.CurrentProfit * 100 if dayTradingPos.CurrentProfit is not None else None
        self.CurrentProfitMonetary = dayTradingPos.CurrentProfitMonetary
        self.CurrentProfitLastTrade  = dayTradingPos.CurrentProfitLastTrade
        self.CurrentProfitMonetaryLastTrade = dayTradingPos.CurrentProfitMonetaryLastTrade
        self.IncreaseDecrease = dayTradingPos.IncreaseDecrease
        self.MaxProfit = dayTradingPos.MaxProfit * 100 if dayTradingPos.MaxProfit is not None else None
        self.MaxLoss = dayTradingPos.MaxLoss * 100 if dayTradingPos.MaxLoss is not None else None

    def toJSON(self):
        return json.dumps(self, default=lambda o: o.__dict__,sort_keys=True)

File: test_clients/C#/day_trader_testclient/DayTraderTestClient/DayTraderTestClient.Common/DTO/Positions/PositionDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DayTraderTestClient.Common.DTO.positions
{
    public class PositionDTO
    {
        public string Symbol { get; set; }

        public int PositionSize { get; set; }

        public bool IsRouting { get; set; }

        public bool LongSignal { get; set; }

        public bool ShortSignal { get; set; }

        public string SignalType { get; set; }

        public string SignalDesc { get; set; }

        public string IsOpen { get; set; }

        public decimal? CurrentMarketPrice { get; set; }

        public decimal? CurrentProfit { get; set; }

        public decimal? CurrentProfitMonetary { get; set; }

        public decimal? CurrentProfitLastTrade { get; set; }

        public decimal? CurrentProfitMonetaryLastTrade { get; set; }

        public decimal? IncreaseDecrease { get; set; }

        public decimal? MaxProxit { get; set; }

        public decimal? MaxLoss { get; set; }

        public int NetShares { get; set; }

        public string Direction { get; set; }

        public string Msg { get; set; }

    }
}
